### PR TITLE
fix compilation and --www, fix crash when modules are in package subdirs

### LINKED ISSUE #51: Assign to each extension of a given module its subdirectories if any exist
This would allow prevention of errors for a given module that has different subdirectories hiding a set of extensions.

### PR DESCRIPTION
**Pull request by RadioPotin**
So this tends to fix #51 and #74 as well as a typo for the launching of `--www` option and some unwanted parenthesis in `src/frontend/searchAdvanced.ml`

### CODE REVIEW

File: src/digodoc_lib/module.ml [hunk 1]
```diff
@@ -20,13 +20,15 @@ let long_name ~mdl_name ~mdl_opam =
   mdl_opam.opam_name ^ "::" ^ mdl_name
 
 let file mdl ~ext =
-  mdl.mdl_dir.dir_name // ( mdl.mdl_basename ^ ext )
+  let name_n_ext = mdl.mdl_basename ^ ext in
+  let path = StringMap.find ext mdl.mdl_path in
+  path // name_n_ext
```
Comment (hernoufM) [inline src/digodoc_lib/module.ml:25]: Don't need to concat this. Values inside ```mdl.mdl_path``` are the paths to the module but not to the directory containig module with given extension.

File: src/digodoc_lib/module.ml [hunk 2]
```diff
@@ -95,5 +95,6 @@ let find_or_create ~mdl_ext ~mdl_dir ~mdl_basename state ~mdl_opam ~objinfo =
   mdl
 
 let file m ext =
```
Comment (hernoufM) [inline src/digodoc_lib/module.ml:97]: Actually, this is a redundant function, better to remove.

File: src/digodoc_lib/odoc.ml [hunk 3]
```diff
@@ -136,7 +137,7 @@ let call_odoc ~continue_on_error state mdl ~pkgs ext =
       "odoc" ; "compile" ;
       "--pkg" ; pkg ;
       "-o" ; odoc_target ;
-      state.opam_switch_prefix // mdl.mdl_dir.dir_name //
+      state.opam_switch_prefix // StringMap.find ext mdl.mdl_path //
```
Comment (hernoufM) [inline src/digodoc_lib/odoc.ml:140]: Same comment. Values inside mdl.mdl_path are the paths to the module.

File: src/digodoc_lib/odoc.ml [hunk 4]
```diff
@@ -170,7 +171,7 @@ let call_odoc_compile ~continue_on_error state mdl ~pkgs ext =
...
```
Comment (hernoufM) [inline src/digodoc_lib/odoc.ml:174]: Same

File: src/digodoc_lib/compute.ml [hunk 5]
```diff
@@ -84,8 +85,8 @@ let check_file state ~objinfo opam_package file =
     | "cmti" ->
         let ( _mdl : ocaml_mdl)  =
           Module.find_or_create state ~objinfo
-            ~mdl_basename:basename ~mdl_ext:ext
-            ~mdl_opam:opam_package ~mdl_dir:dir
+            ~mdl_basename:basename ~mdl_ext:("." ^ ext)
```
[the remark this answers is not in the document]
Comment (RadioPotin) [reply src/digodoc_lib/compute.ml:88]: When I remove it i get a Not_Found error...

[the remark this answers is not in the document]
Comment (hernoufM) [reply src/digodoc_lib/compute.ml:88]: Where does it appears, Not_found?

**Pull request merged**